### ISSUE TITLE
Question: I2Pd and SOCKS

### ISSUE THREAD

**Issue by o0nd7ots** [PurpleI2P/i2pd#1986]
What versions of SOCKS does I2Pd support, both for staying within the anonymity network and outproxy?

The default config leads be to believe that for outproxy only 4 is supported (not even 4a).
The issues make me believe that for internal traffix 5 is okay.
I did not know where to look for documentation for this.

**Comment by o0nd7ots** [comment]
Thanks. Do you know about outproxy traffic and v4 for internal?

Also, where could I find this?

**Comment by Vort** [comment]
I2P do not have IP addresses for internal resources.
That's why SOCKS 4 (without a) makes no sense.

> Do you know about outproxy traffic

I did not tested outproxies.
But I heard that developers want to fix SOCKS support for them.
It's like plans for indefinite future.

**Comment by o0nd7ots** [comment]
@orignal 
Thanks. Does 4a work?

**Comment by orignal** [comment]
I don't know. Better to ask @majestrate

**Comment by majestrate** [comment]
> @orignal Thanks. Does 4a work?

probably.

**Comment by o0nd7ots** [comment]
If this is the best we can get then this issue can be closed.
I will try to work that info in to the documentation/readme.

**Comment by Alfa21tpp** [comment]
if it can be of any use, in my privoxy configuration I use this:
`forward-socks4a .i2p 127.0.0.1:4447 .`
so v4a is verified, imo

**Comment by orignal** [comment]
Yes, incoming socks4 works, the recent change is about upstream proxy only
